implementation/contracts/test/keep/ECDSAKeepStub.sol

```diff
@@ -1,74 +1,96 @@
 pragma solidity ^0.5.10;
 
-import {IBondedECDSAKeep} from "@keep-network/keep-ecdsa/contracts/api/IBondedECDSAKeep.sol";
+import {
+    IBondedECDSAKeep
+} from "@keep-network/keep-ecdsa/contracts/api/IBondedECDSAKeep.sol";
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 
 /// @notice Implementation of ECDSAKeep interface used in tests only
 /// @dev This is a stub used in tests, so we don't have to call actual ECDSAKeep
 contract ECDSAKeepStub is IBondedECDSAKeep {
+    using SafeMath for uint256;
+
     bytes publicKey;
     bool success;
+    uint256 memberCount = 3;
+    bool isActive = true;
     uint256 bondAmount = 10000;
 
     // Notification that the keep was requested to sign a digest.
-    event SignatureRequested(
-        bytes32 digest
-    );
+    event SignatureRequested(bytes32 digest);
 
     // Define fallback function so the contract can accept ether.
     function() external payable {}
 
+    // Functions implemented for IBondedECDSAKeep interface.
+
     function getPublicKey() external view returns (bytes memory) {
         return publicKey;
     }
 
-    function sign(bytes32 _digest) external {
-        emit SignatureRequested(_digest);
+    function checkBondAmount() external view returns (uint256) {
+        return bondAmount;
     }
 
-    function returnPartialSignerBonds() external payable {
-    // solium-disable-previous-line no-empty-blocks
+    function sign(bytes32 _digest) external onlyWhenActive {
+        emit SignatureRequested(_digest);
     }
 
-    function distributeETHToMembers() external payable {
-    // solium-disable-previous-line no-empty-blocks
+    function distributeETHReward() external payable {
+        uint256 dividend = msg.value.div(memberCount);
+
+        require(dividend > 0, "Dividend value must be non-zero");
```

Comment:
Hmmm… I'm 50-50 on this change. At this point this ceases to be a stub. Plus we don't end up correctly testing the interface; instead, we start integration testing the stub :/

The current situation is imperfect, but I think we should stick with what we have and work on system and integration tests out of band to catch the broader issue.

Thoughts?

Reply:
I'm not a huge fan of this solution either. I wanted to have anything to catch possible integration problems sooner than later. This is definiately a case for integration tests.

Reply:
Yeah, I think I'm interested in just leaving this. We would fail to catch the next integration problem unless the change made in keep-ecdsa was made to this stub as well, which is roughly equivalent to having someone on keep-ecdsa also reviewing tbtc ecdsa-related changes, which is what we should have done in the first place when this originally happened (and which I bypassed).

I think I'd rather leave the stub as-is and pull in just the fix, tbh :/ 

Also I imagine since the diff didn't include it that this was clear, but we shouldn't bump the keep-ecdsa dependency to -rc in a PR to master :)

Reply:
> Also I imagine since the diff didn't include it that this was clear, but we shouldn't bump the keep-ecdsa dependency to -rc in a PR to master :)

We didn't release the `0.10.0` version, but we probably should - CC @pdyraga.

Reply:
We did. Please see:
https://github.com/keep-network/keep-ecdsa/packages/29040?version=0.10.0-rc
https://github.com/keep-network/keep-ecdsa/pull/310

Reply:
We did release `0.10.0-rc` but I believe @Shadowfiend expected `0.10.0` to be used in `tbtc` dependencies. Am I correct?

Reply:
We should reference `0.10.0-pre`, as here: https://github.com/keep-network/tbtc.js/blob/master/package.json#L26

However, I just realized that until I finish all my packaging stuff we may not be publishing `-pre` versions, so… Whatever works, works.